optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/IntervalJoiners.java

```diff
@@ -0,0 +1,314 @@
+     * @param <E> The type of the fifth argument
+     * @param <Property_> The type used to define the interval, comparable
+     * 
+     * @param leftStartMapping Maps the first, second, third and fourth arguments to the first interval start point
+     * @param leftEndMapping Maps the first, second, third and fourth arguments to the first interval end point
+     * @param rightStartMapping Maps the fifth argument to the second interval start point
+     * @param rightEndMapping Maps the fifth argument to the second interval end point
+     * 
+     * @return An indexed joiner that filters the constraint stream to only include
+     *         elements (A,B,C,D,E) where (A,B,C,D)'s and E's intervals (as defined by the function mapping)
+     *         overlap.
+     */
+    public static <A, B, C, D, E, Property_ extends Comparable<Property_>> PentaJoiner<A, B, C, D, E> overlaps(
+            QuadFunction<A, B, C, D, Property_> leftStartMapping,
+            QuadFunction<A, B, C, D, Property_> leftEndMapping,
+            Function<E, Property_> rightStartMapping,
+            Function<E, Property_> rightEndMapping) {
+        return AbstractPentaJoiner.merge(Joiners.lessThan(leftStartMapping, rightEndMapping),
+                Joiners.greaterThan(leftEndMapping, rightStartMapping));
+    }
+
+    /**
+     * Like {@link #during(Function, Function, Function, Function)}, but for
+     * PentaConstraintStream.
+     * 
+     * @param <A> The type of the first argument
+     * @param <B> The type of the second argument
+     * @param <C> The type of the third argument
+     * @param <D> The type of the fourth argument
+     * @param <E> The type of the fifth argument
+     * @param <Property_> The type used to define the interval, comparable
+     * 
+     * @param leftStartMapping Maps the first, second, third and fourth arguments to the first interval start point
+     * @param leftEndMapping Maps the first, second, third and fourth arguments to the first interval end point
+     * @param rightStartMapping Maps the fifth argument to the second interval start point
+     * @param rightEndMapping Maps the fifth argument to the second interval end point
+     * 
+     * @return An indexed joiner that filters the constraint stream to only include
+     *         elements (A,B,C,D,E) where (A,B,C,D)'s interval contains E's interval (as defined by the function mapping)
+     */
+    public static <A, B, C, D, E, Property_ extends Comparable<Property_>> PentaJoiner<A, B, C, D, E> during(
+            QuadFunction<A, B, C, D, Property_> leftStartMapping,
+            QuadFunction<A, B, C, D, Property_> leftEndMapping,
+            Function<E, Property_> rightStartMapping,
+            Function<E, Property_> rightEndMapping) {
+        return AbstractPentaJoiner.merge(Joiners.lessThanOrEqual(leftStartMapping, rightStartMapping),
+                Joiners.greaterThanOrEqual(leftEndMapping, rightEndMapping));
```

Comment:
In fact, now that I see how little code there actually is in this class, makes me think that maybe `IntervalJoiners` class is unnecessary and the methods could just be inlined into `Joiners`?

When I was making this suggestion, I was somehow expecting that there would be more code here.

Reply:
Probably from the 13 cases required to ensure full checking of joiner correctness (`4! = 24`, but pairs (a,b) where a > b are invalid, which cut it in half, and one additional case for equals). Simple code may not have so simple tests.